Fix unit tests on windows

rata opened the pull request:
Let's provide a stub implementation for windows, so tests pass fine.

This is an alternative to: https://github.com/kubernetes/kubernetes/pull/130794. This is the preferred PR. 

cc @haircommander @mrunalp @giuseppe 

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind bug
/kind failing-test

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

Windows unit test seem to be broken on master, this PR fixes it.
It wasn't detected before because CI was not running unit tests for the kubelet on windows (not even optional, it seems?), it was just enabled.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

rata committed:
pkg/kubelet: Fix userns tests on Windows

Signed-off-by: Rodrigo Campos <rodrigoca@microsoft.com>

rata committed:
pkg/kubelet/userns: Provide stub implementation for windows

Signed-off-by: Rodrigo Campos <rodrigoca@microsoft.com>

rata committed:
pkg/kubelet/userns: Wrap error to get mappings

I needed to wrap the error for debugging, let's just keep this as it is
useful.

Signed-off-by: Rodrigo Campos <rodrigoca@microsoft.com>

marosset commented:
/test pull-kubernetes-windows-unit

marosset commented:
/kind bugfix

marosset commented:
/kind regression

marosset commented:
code looks good to me.
[pull-kubernetes-unit-windows-master](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/130800/pull-kubernetes-unit-windows-master/1900264120104521728) should confirm the issue is fixed too

marosset commented:
/lgtm
thanks for the quick turn-around on this fix @rata

giuseppe reviewed (approved):
LGTM

rata committed:
pkg/kubelet/userns: Remove skip on windows

We don't build these tests for Windows, let's remove this skip.

We should have never added that skip, we should have skipped the entire
suite on Windows.

Signed-off-by: Rodrigo Campos <rodrigoca@microsoft.com>

marosset commented:
/lgtm

BenTheElder commented on pkg/kubelet/userns/types.go:
```diff
@@ -0,0 +1,13 @@
+package userns
```
This is missing the copyright header, which is failing CI:
https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/130800/pull-kubernetes-verify/1900264114270244864

rata commented:
@BenTheElder fixed, thanks!

marosset commented:
Windows nodes came up successfully in `pull-kubernetes-e2e-capz-windows-master`.
The failure in there looks both new and unrelated to this change.

rata commented:
Cool, one more test (not sure if it's required) and a lgtm and this should be merged :)

rata commented:
All green but that, @marosset can you lgtm? It seems it's missing only that

marosset commented:
/lgtm

rata commented:
Created a simple follow-up to simplify the code slightly: #130820
